ci/create-tag-file.yml

```diff
@@ -0,0 +1,79 @@
+---
+#
+# (c) Copyright IBM Corp. 2021
+# (c) Copyright Instana Inc.
+#
+
+# see https://concourse-ci.org/tasks.html
+platform: linux
+image_resource:
+  type: registry-image
+  source:
+    repository: registry.access.redhat.com/ubi8/ubi
+outputs:
+  - name: message
+  - name: version
+params:
+  COMMIT_SHA: ((commit-sha))
+  CLASSIFIER: ((classifier))
+  VERSION: ((version))
+  FLAVOR: ((flavor))
+  RELEASE_TREE: ((release-tree))
+run:
+  path: /bin/bash
+  args:
+  - -exc
+  - |
+      set -e
+      # CLASSIFIER is used to distiguish OpenJDK (classic) agent images and OpenJ9 agent images. OpenJ9 images will use the CLASSIFIER "-j9".
+      # Here we fix it in case only 'j9' is given.
+      # Empty CLASSIFIER means classic OpenJDK based images.
+
+      # NOTE: for s390x, ppc64le J9 is the default runtime, RPM/DEB packages for s390x, ppc64le with Azul 8 Java are unavailable
+
+      # CLASSIFIER is not empty and does not start with a '-':
+      if [ "${CLASSIFIER}x" != 'x' ] && [ "${CLASSIFIER#-}" = "${CLASSIFIER}" ]; then
+        CLASSIFIER="-${CLASSIFIER}"
+      fi
+
+      if [ "${RELEASE_TREE}x" = 'x' ]; then
+        RELEASE_TREE="prerelease"
+      fi
+
+      # Only tag with commit SHA on pre-release:
+      COMMIT_SHA_TAG=""
+      if [ "${RELEASE_TREE}x" = 'prereleasex' ]; then
```

Comment:
I think here you don't need to add the `x`. Just `"${RELEASE_TREE}" = 'prerelease'` would be more readable.

In the other places the addition of the `x` is only to handle both empty and not-set variables correctly.